neuralmonkey/encoders/cnn_encoder.py

```diff
@@ -141,17 +197,172 @@ def output(self) -> tf.Tensor:
             dropout_keep_prob=self.dropout_keep_prob,
             train_mode=self.train_mode)
 
-    @tensor
-    def spatial_mask(self) -> tf.Tensor:
-        return tf.ones(tf.shape(self.spatial_states)[:3])
-
     def feed_dict(self, dataset: Dataset, train: bool = False) -> FeedDict:
         # if it is from the pickled file, it is list, not numpy tensor,
         # so convert it as as a prevention
         images = np.array(dataset.get_series(self.data_id))
 
         f_dict = {}
         f_dict[self.image_input] = images / 225.0
+        # it is one everywhere where non-zero, i.e. zero columns are masked out
```

Comment:
Tady bych `it` nahradil vhodnym objektem - `image mask`? a specifikoval bych, kam koukáme na ty `non-zero`s - `images`?

Jsou opravdu výsledkem sumy signů jedničky? Nemělo by se to ještě něčim obalit? Třeba dalším signem?

Reply:
Je to blbě, má to být obráceně, ale protože jsem to krokoval na černobílých obrázcích, tak jsem si toho nevšim.